src/main/java/frc/robot/subsystems/drivetrain/commands/OverpoweredDrive.java

```diff
@@ -0,0 +1,105 @@
+package frc.robot.subsystems.drivetrain.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Robot;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Constants.SwerveDrive.VELOCITY_MULTIPLIER;
+
+public class OverpoweredDrive extends HolonomicDrive {
+    PIDController pidController = new PIDController(0.03, 0, 0) {{
+        enableContinuousInput(-180, 180);
+        setTolerance(5);
+    }};
+    private boolean newSetpoint = false;
+    private Rotation2d setpoint;
+    private boolean wait = false;
+    private double current = 0;
+    private final Timer timer = new Timer();
+
+    public OverpoweredDrive(SwerveDrive swerveDrive, DoubleSupplier forwardSupplier, DoubleSupplier strafeSupplier, DoubleSupplier rotationSupplier) {
+        super(swerveDrive, forwardSupplier, strafeSupplier, rotationSupplier);
+    }
+
+    @Override
+    public void execute() {
+        ChassisSpeeds speeds = calculateVelocities();
+        double forward = speeds.vxMetersPerSecond;
+        double strafe = speeds.vyMetersPerSecond;
+        double magnitude = Math.hypot(forward, strafe);
+        double alpha = Math.atan2(strafe, forward);
+        if (magnitude == 0) current = 0;
+        current += magnitude / 10;
+        if (current > magnitude) current = magnitude;
+        forward = Math.cos(alpha) * current;
+        strafe = Math.sin(alpha) * current;
+        double rotation = speeds.omegaRadiansPerSecond;
```

Comment:
I would extract it to another method (or even extend the `calculateVelocities()` method)